### PR TITLE
CMake support

### PR DESCRIPTION
**Pull request by isaachier**

### CODE REVIEW

**Review by ruslo — CHANGES_REQUESTED** [review]
Same notes as in https://github.com/hunter-packages/leveldb/pull/2

**Review by ruslo — CHANGES_REQUESTED** [review]
Let's finish https://github.com/hunter-packages/leveldb/pull/2 first. Both have same issues.

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,362 @@
+# Copyright 2017 The LEVELDB Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+cmake_minimum_required(VERSION 3.1)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
```
Comment (ruslo) [inline CMakeLists.txt]: `CMAKE_SOURCE_DIR` is root directory of the sources. If this project will be used as a subproject then `${CMAKE_SOURCE_DIR}/cmake` will not mean directory relative to this file. See:
* https://cgold.readthedocs.io/en/latest/tutorials/cmake-sources/includes.html#cmake-current-list-dir-vs-cmake-current-source-dir

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+set(CMAKE_TOOLCHAIN_FILE "${CMAKE_SOURCE_DIR}/cmake/toolchain.cmake")
```
Comment (ruslo) [inline CMakeLists.txt]: Set it to cache so user can customize it:
* https://cgold.readthedocs.io/en/latest/tutorials/variables/cache.html#use-case

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+)
+
+# TODO(pwnall): This is only needed for port_posix.
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+target_link_libraries(leveldb Threads::Threads)
+set(deps Threads)
+
+if(LEVELDB_WITH_CRC32C)
+  target_link_libraries(leveldb crc32c::crc32c)
```
Comment (ruslo) [inline CMakeLists.txt]: I guess it's public dependency, should be `target_link_libraires(... PUBLIC ...)`.

File: cmake/toolchain.cmake [hunk 4]
```diff
@@ -0,0 +1,3 @@
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED OFF)
```
Comment (ruslo) [inline cmake/toolchain.cmake]: Why `OFF`? I guess it should be `ON`.

**Review by ruslo — CHANGES_REQUESTED** [review]

File: cmake/toolchain.cmake (hunk 4 above)
Comment (isaachier) [reply cmake/toolchain.cmake]: Project owner wanted to use C++11 when  present. Isn't strictly required.

File: CMakeLists.txt (hunk 1 above)
Comment (isaachier) [reply CMakeLists.txt]: OK

**Review by ruslo — APPROVED** [review]

**Pull request merged**